src/utilities/new_versions.jl

```diff
@@ -99,3 +99,213 @@ function skip_equality_specifiers(
            !contains(compat_entry_verbatim, '>') &&
            !contains(compat_entry_verbatim, '<')
 end
+
+function create_new_pull_request(
+    api::GitLab.GitLabAPI,
+    repo::Gitlab.Project,
+    new_branch_name::AbstractString,
+    master_branch_name::AbstractString,
+    title::AbstractString,
+    body::AbstractString,
+)
+    return create_pull_request(
+        api,
+        repo.owner.name,
+        repo.name;
+        id=repo.id,
+        source_branch=new_branch_name,
+        target_branch=master_branch_name,
+        title=title,
+        body=body,
+    )
+end
+
+function create_new_pull_request(
+    api::GitHub.GutHubAPI,
+    repo::GitHub.Repo,
+    new_branch_name::AbstractString,
+    master_branch_name::AbstractString,
+    title::AbstractString,
+    body::AbstractString,
+)
+    return create_pull_request(
+        api,
+        repo.owner.login,
+        repo.name;
+        id=repo.id,
+        source_branch=new_branch_name,
+        target_branch=master_branch_name,
+        title=title,
+        body=body,
+    )
+end
+
+function get_url_with_auth(
+    api::GitHub.GitHubAPI, hostname::AbstractString, repo::GitHub.Repo
+)
+    return "https://x-access-token:$(api.token.token)@$hostname/$(repo.full_name).git"
+end
+
+function get_url_for_ssh(::GitHub.GitHubAPI, hostname::AbstractString, repo::GitHub.Repo)
+    return "git@$hostname:$(repo.full_name).git"
+end
+
+function get_url_with_auth(
+    api::GitLab.GitLabAPI, hostname::AbstractString, repo::GitLab.Project
+)
+    return "https://oauth2:$(api.token.token)@$hostname/$(repo.path_with_namespace).git"
+end
+
+function get_url_for_ssh(::GitLab.GitLabAPI, hostname::AbstractString, repo::GitLab.Project)
+    return "git@$(hostname):$(repo.path_with_namespace).git"
+end
+
+function make_pr_for_new_version(
+    forge::Forge,
+    clone_hostname::AbstractString,
+    repo::Union{GitHub.Repo,GitLab.Project},
+    dep::DepInfo,
+    entry_type::EntryType,
+    ci_cfg::CIService,
+    new_compat_entry::String;
+    subdir::AbstractString="",
+    master_branch::Union{DefaultBranch,AbstractString},
+    env::AbstractDict=ENV,
+    bump_compat_containing_equality_specifier::Bool=true,
+)
+    # Determine if we need to make a new PR
+    if (!isnothing(dep.latest_version) && !isnothing(dep.version_spec)) &&
```

Comment:
Can we not just do `dep.latest_version in dep.version_spec`?

Reply:
I'm not 100% show how all that stuff was changed. I see they are union with Nothing, are they guaranteed to have something in them at this point?

Reply:
So `dep.version_spec` can be `nothing` if the `Project.toml` does not contain a version for the dependency. However, we can still drop the `!isnothing(dep.latest_version)` safely as you can check that `nothing in <:Collection`:

```julia
julia> nothing in [1,2,3]
false
```

But we still want to check the `!isnothing(dep.version_spec)` since,

```julia
julia> 1 in nothing
ERROR: MethodError: no method matching iterate(::Nothing)
Closest candidates are:
  iterate(::Union{LinRange, StepRangeLen}) at range.jl:664
  iterate(::Union{LinRange, StepRangeLen}, ::Int64) at range.jl:664
  iterate(::T) where T<:Union{Base.KeySet{var"#s79", var"#s78"} where {var"#s79", var"#s78"<:Dict}, Base.ValueIterator{var"#s77"} where var"#s77"<:Dict} at dict.jl:693
  ...
Stacktrace:
 [1] in(x::Int64, itr::Nothing)
   @ Base ./operators.jl:1131
 [2] top-level scope
   @ REPL[2]:1
```